gateway/ln-gateway/proto/gatewaylnrpc.proto

```diff
@@ -38,12 +40,49 @@ message PayInvoiceResponse {
   bytes preimage = 1;
 }
 
-message SubscribeInterceptHtlcsRequest {
-  // Subscribe to intercepted HTLCs with this short channel id
-  uint64 short_channel_id = 1;
+message InterceptHtlcsRequest {
+  message Subscribe {
+    // The short channel id of HTLCs to intercept
+    uint64 short_channel_id = 1;
+  }
+
+  message Settle {
+    // The preimage for settling an intercepted HTLC
+    bytes preimage = 1;
+  }
+
+  message Cancel {
+    // The reason for the cancellation of an intercepted HTLC
+    string reason = 1;
+  }
+
+  oneof action {
+    /// Request to subscribe to HTLCs with a specific short channel id
+    ///
+    /// Send this request when the gateway just assigned a new channel id to a newly
+    /// connected federation. GatewayLightning will respond with a stream of
+    /// intercept HTLCs for the gateway to process
+    Subscribe subscribe = 1;
+
+    /// Request to complete an intercepted HTLC with success result after
+    /// processing
+    ///
+    /// Send this request when the gateway successfully processed intercepted HTLC
+    /// GatewayLightning will settle/resolve the intercepted HTLC with reason
+    /// provided.
+    Settle settle = 2;
+
+    /// Request to complete an intercepted HTLC with failure result after
+    /// processing
+    ///
+    /// Send this request when the gateway failed or canceled processing of
+    /// intercepted HTLC. GatewayLightning will fail/cancel the intercepted HTLC
+    /// with reason provided.
+    Cancel cancel = 3;
+  }
 }
 
-message SubscribeInterceptHtlcsResponse {
+message InterceptedHtlcResponse {
```

Comment:
This type's docs seem to talk a lot about invoices, but intercepted HTLCs don't tell you much about the invoice from which it was generated. 

Also the `units` member isn't all that useful imo. Just make it msat? Or is this a LND thing we try to be compatible with?

Reply:
> This type's docs seem to talk a lot about invoices, but intercepted HTLCs don't tell you much about the invoice from which it was generated.
> Also the units member isn't all that useful imo. Just make it msat? Or is this a LND thing we try to be compatible with?

Update to require interceptors send amounts in msat. Both LND and CLN can do this today